acoustid/future/fpindex/updater.py

```
@@ -0,0 +1,324 @@
+            if loaded_count % 1000 == 0:
+                logger.info(
+                    "Processed %d/%d fingerprints (%.2f%%)",
+                    loaded_count,
+                    total_count,
+                    (loaded_count / total_count * 100),
+                )
+                if shutdown_event.is_set():
+                    logger.info("Interrupting initial data load due to shutdown event")
+                    raise ShutdownError
+
+    logger.info(f"Initial data load complete. Processed {loaded_count} fingerprints.")
```

Ruff (0.8.2)

Logging statement uses f-string

(G004)